app/src/features/FhirResourceTree/resourceTreeUtils.ts

```diff
@@ -89,14 +89,17 @@ export const createElementNode = (
   const elementPath = `${prefixPath}${
     prefixPath && suffixPath && "."
   }${suffixPath}${index !== undefined ? `[${index}]` : ""}`;
+  const elementName = `${elementDefinition.id?.split(".").pop() ?? ""}${
+    elementDefinition.sliceName ? `:${elementDefinition.sliceName}` : ""
```

Comment:
I wanted to tell you to refactor this mess but actually I couldn't find a better way myself. I was thinking of using dedicated functions to compute these strings but in the end I don't think it would be clearer.